File: src/main/java/com/soukhyatech/faceattendance/controller/StatsController.java
package com.soukhyatech.faceattendance.controller;

import com.soukhyatech.faceattendance.model.Employee;
import com.soukhyatech.faceattendance.service.AttendanceService;
import com.soukhyatech.faceattendance.service.EmployeeService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.LinkedHashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/stats")
@RequiredArgsConstructor
public class StatsController {

    private final EmployeeService employeeService;
    private final AttendanceService attendanceService;

    @GetMapping
    @PreAuthorize("hasAnyRole('ADMIN','HR')")
    public ResponseEntity<Map<String, Object>> getStats() {
        long totalEmployees = employeeService.count();
        long totalRecords = attendanceService.count();
        var today = attendanceService.getTodayStats();
        long active = employeeService.countByStatus(Employee.Status.Active);
        long hibernate = employeeService.countByStatus(Employee.Status.Hibernate);
        long onLeave = employeeService.countByStatus(Employee.Status.On_Leave);
        long resigned = employeeService.countByStatus(Employee.Status.Resigned);

        long activePercent = totalEmployees > 0 ? Math.round((double) active / totalEmployees * 100) : 0;
        long hibernatePercent = totalEmployees > 0 ? Math.round((double) hibernate / totalEmployees * 100) : 0;

        Map<String, Object> res = new LinkedHashMap<>();
        res.put("success", true);
        res.put("total_employees", totalEmployees);
        res.put("present_today", today.getPresentToday());
        res.put("late_today", today.getLateToday());
        res.put("total_records", totalRecords);
        res.put("status_counts", Map.of(
            "active", active,
            "hibernate", hibernate,
            "on_leave", onLeave,
            "resigned", resigned
        ));
        res.put("active_percent", activePercent);
        res.put("hibernate_percent", hibernatePercent);
        res.put("dept_hibernate_counts", employeeService.getDeptHibernateCounts());
        res.put("monthly_hibernate_trend", employeeService.getMonthlyHibernateTrends());
        return ResponseEntity.ok(res);
    }
}

File: csharp/SoukhyaTech.FaceAttendance/Controllers/StatsController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using SoukhyaTech.FaceAttendance.Repositories;

namespace SoukhyaTech.FaceAttendance.Controllers
{
    [ApiController]
    [Route("api/stats")]
    [Authorize(Roles = "ADMIN,HR")]
    public class StatsController : ControllerBase
    {
        private readonly IEmployeeRepository _employeeRepo;
        private readonly IAttendanceRepository _attendanceRepo;

        public StatsController(IEmployeeRepository employeeRepo, IAttendanceRepository attendanceRepo)
        {
            _employeeRepo = employeeRepo; _attendanceRepo = attendanceRepo;
        }

        [HttpGet]
        public async Task<IActionResult> GetStats()
        {
            long totalEmployees = await _employeeRepo.CountAsync();
            long totalRecords = await _attendanceRepo.CountAsync();
            var today = DateTime.UtcNow.ToString("yyyy-MM-dd");
            var (presentToday, lateToday) = await _attendanceRepo.GetTodayStatsAsync(today + "T00:00:00", today + "T23:59:59.999");
            long active = await _employeeRepo.CountByStatusAsync("Active");
            long hibernate = await _employeeRepo.CountByStatusAsync("Hibernate");
            long onLeave = await _employeeRepo.CountByStatusAsync("On Leave");
            long resigned = await _employeeRepo.CountByStatusAsync("Resigned");

            return Ok(new
            {
                success = true,
                total_employees = totalEmployees,
                present_today = presentToday,
                late_today = lateToday,
                total_records = totalRecords,
                status_counts = new { active, hibernate, on_leave = onLeave, resigned },
                active_percent = totalEmployees > 0 ? Math.Round((double)active / totalEmployees * 100) : 0,
                hibernate_percent = totalEmployees > 0 ? Math.Round((double)hibernate / totalEmployees * 100) : 0,
                dept_hibernate_counts = await _employeeRepo.GetDeptHibernateCountsAsync(),
                monthly_hibernate_trend = await _employeeRepo.GetMonthlyHibernateTrendsAsync()
            });
        }
    }
}